src/Count.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+/*
+ * This file is part of GitHub Profile Views Counter.
+ *
+ * (c) Anton Komarev <anton@komarev.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Komarev\GitHubProfileViewsCounter;
+
+use Webmozart\Assert\Assert;
+
+final class Count
+{
+    /**
+     * Wrong behavior around extremely large values if set
+     * to an extremely large value (near PHP_INT_MAX)
+     * Shouldn't exceed PHP_INT_MAX to avoid casting to a float
+     */
+    private const MAX_COUNT = PHP_INT_MAX;
+
+    private int $count;
+
+    public function __construct(
+        float $count
+    ) {
+        Assert::lessThan(
+          $count,
+          self::MAX_COUNT,
+          'The maximum number of views has been reached'
+        );
+        $this->count = intval($count);
+    }
+
+    public static function ofString(string $countStr): self
+    {
+      Assert::digits(
+        $countStr,
+        'The base count must be a positive integer'
```

Comment:
Not sure that digits checks for positive value:

```php
    public static function digits($value, $message = '')
    {
        $locale = \setlocale(LC_CTYPE, 0);
        \setlocale(LC_CTYPE, 'C');
        $valid = !\ctype_digit($value);
        \setlocale(LC_CTYPE, $locale);

        if ($valid) {
            static::reportInvalidArgument(\sprintf(
                $message ?: 'Expected a value to contain digits only. Got: %s',
                static::valueToString($value)
            ));
        }
    }
```

Reply:
this method should not return error accordingly baseCount, because it may be called in other cases in the future

Reply:
`-` is considered a non-digit. So it returns false for negative values.

Reply:
> this method should not return error accordingly baseCount, because it may be called in other cases in the future

So we should do a positive values check in the constructor as well?

Reply:
I think that only check in constructor is enough

Reply:
Wouldn't we still need a check in the `ofString` method to ensure no non-digits are passed?